pkg/util/files.go

```diff
@@ -98,28 +93,49 @@ func Untar(source string, dest string) error {
 			break
 		}
 		if err != nil {
-			return err
+			return fmt.Errorf("Error during read of tar archive %v, err: %v", source, err)
 		}
 
-		if file.Typeflag == tar.TypeDir {
-			err = os.Mkdir(filepath.Join(dest, file.Name), 0755)
-			if err != nil {
-				return err
+		// If we have an extractionDir and this doesn't match, skip it.
+		if !strings.HasPrefix(file.Name, extractionDir) {
+			continue
+		}
+		// Now transform the filename to skip the extractionDir
+		file.Name = strings.TrimPrefix(file.Name, extractionDir)
+
+		// If file.Name is now empty this is the root directory we want to extract, and need not do anything.
+		if file.Name == "" && file.Typeflag == tar.TypeDir {
+			continue
+		}
+
+		fullPath := filepath.Join(dest, file.Name)
+
+		// At this point only directories and block-files are handled. Symlinks and the like are ignored.
```

Comment:
I checked this out, and what happens is when treated as an ordinary file a symlink comes in as a zero-length file with the correct name, but no other information, which is what I'd expect, since a symlink in a Unix-type fs is really just a file with the path to another file in it. It's true that you can get tar to turn symlinks into files (following them) when you're *creating* a tarball, but doing it at extraction time would 1) only work for internal, relative references and 2) require creating an index of all the paths in the tarball, 3) resolve symlinks after everything had been traversed.  I don't think it's likely worth the effort. 

Demonstration in https://github.com/rfay/ddev/tree/20170512_demonstrate_symlink_tar and there's a tarball download prebuilt there as well.  Just the commit in https://github.com/rfay/ddev/commit/12e94cb92ec9b4fc28af4ab6090a3b4ec597133a shows the idea.

It seems like maybe this is what you were saying in your comment; however, I'd probably be  happier just not creating files that are broken. It *is* possible we need to do more here.